Gems/WhiteBox/Code/Source/EditorWhiteBoxComponent.cpp

```diff
@@ -318,6 +317,7 @@ namespace WhiteBox
         if (AzToolsFramework::IsEntityVisible(entityId))
         {
             ShowRenderMesh();
```

Comment:
I know this existed prior to your review, but the EditorWhiteBoxComponent's ShowRenderMesh and HideRenderMesh look a bit heavy handed, nullifying the render mesh in order to hide it. I think using m_renderMesh->SetVisibility would be better there now that it has been implemented. 

Otherwise you have three different (potentially competing) states to determine visibility:
1. Is EditorWhiteBoxComponent::m_renderMesh null
2. Is EditorWhiteBoxComponent::m_renderData.m_material.m_visible true
3. Is AtomRenderMesh::m_visible true